xcm/src/v0/traits.rs

```diff
@@ -67,3 +67,21 @@ impl SendXcm for () {
 		Err(Error::Unimplemented)
 	}
 }
+
+pub trait ExecuteHrmp {
+	fn hrmp_init_open_channel(sender: u32, recipient: u32, max_message_size: u32, max_capacity: u32) -> Result;
+	fn hrmp_accept_open_channel(recipient: u32, sender: u32) -> Result;
+	fn hrmp_close_channel(initiator: u32, sender: u32, recipient: u32) -> Result;
```

Comment:
DQ: why are all parameters u32 all across the place?

Reply:
I presumed it's the same reason why we use u32 within XCM definitions. The alternative you are thinking of is `ParaId`. That comes from the primitives crate. Unlike one may think, it can be pretty heavy, bringing all kinds of cruft. At the same time, we want this crate to be used pervasively thus we want to minimize the number of dependencies.